hazelcast-sql/src/main/java/com/hazelcast/sql/impl/calcite/validate/types/HazelcastOperandTypes.java

```diff
@@ -65,11 +66,11 @@ public static SqlOperandTypeChecker notAllNull(SqlOperandTypeChecker base) {
         return new NotAllNull(base);
     }
 
-    private static final class NotAny implements SqlSingleOperandTypeChecker {
+    private abstract static class DisallowedTypeOperandTypeChecker implements SqlSingleOperandTypeChecker {
```

Comment:
What's the reasoning behind introducing an abstract class with a single non-abstract implementation?

Reply:
A leftover. I was experimenting with some advanced type checking for temporal types but ultimately removed it. Forgot to roll back the changes.
Rolled back in https://github.com/hazelcast/hazelcast/pull/17395/commits/e442d1cae09e167e80a20d5656c891458e84aaae